casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -623,18 +623,9 @@ object BlockAPI {
   )(implicit casper: MultiParentCasper[F]): F[A] =
     for {
       dag <- casper.blockDag
-      // TODO this is temporary solution to not calculate fault tolerance for old blocks which is costly
-      oldBlock  = dag.latestBlockNumber.map(_ - block.body.state.blockNumber).map(_ > 100)
-      isInvalid = dag.lookup(block.blockHash).map(_.get.invalid)
-      // Old block fault tolerance / invalid block has -1.0 fault tolerance
-      oldBlockFaultTolerance = isInvalid.map(if (_) -1f else 1f)
-      normalizedFaultTolerance <- oldBlock.ifM(
-                                   oldBlockFaultTolerance,
-                                   SafetyOracle[F]
-                                     .normalizedFaultTolerance(dag, block.blockHash)
-                                 )
-      initialFault   <- casper.normalizedInitialFault(ProtoUtil.weightMap(block))
-      faultTolerance = normalizedFaultTolerance - initialFault
+      // TODO this is temporary solution to not calculate fault tolerance all the blocks
+      isFinalized    <- dag.isFinalized(block.blockHash)
+      faultTolerance = if (isFinalized) 1f else -1f
```

Comment:
I think we can calculate fault tolerance for non-finalized blocks. This calculation should be the most few hundred milliseconds, right?

Reply:
not for very old blocks

Reply:
Calculating fault tolerance in single-parent mode does not actually have to much meaning

Reply:
After another thinking, we can calculate in the latest 100 block like we did before.